### ISSUE TITLE
Events Track

### ISSUE THREAD

**Issue by DeRobyJ** [LMMS/lmms#2882]
This is just a bunch of ideas for 1.3+.
Differently from my past enhancement request issues, in my opinion this one would be not really hard to program, because it would re-use a lot of funcions already working for mouse clicks events, and could be a potential solution to different problems, like some highly requested features, plus it would allow new ways of music composing.

The hard part is here:
The Events Track would be really similar to an Instrument Track on the Song Editor.
It's be able to store 128 user-defined events, accessible throught MIDI-in and Piano-roll.
(it should not be added, since it's useless to have multiple instances, it should just be shown/minimized in the Sond Editor, eventually)

It indeed has a kind of piano-roll, the "event-roll". It should only have the record button, add/delete/select and cut/copy/paste, then the pianoroll table and the Velocity editor (which will become "Probability, chance" editor, 1 to 100%)

The events are defined in a list (similar to the FX tab on instruments), accessible via the Events Track window.

Last difficult thing, it would need a handler.
___________________________________________________________
Now for the easy and creative things: the kinds of events.

First kind: **play _x_ instrument in BBeditor**

Right now, with mouse clicks, you're able to play a loaded track by clicking the flashing button.
These events would call just the same method.

This will allow to have a drum sampler, mostly wanted for live usage, but also for easier composition.
(moreover, with the "probability" editor in the event-roll, you can have randomized variations)

Second kind: **move loop-point to _x_ position**

The most basic usage would be to move the End-point to a new section, wait for the cursor to enter it, and then move the Start-point. 
Having this Event Track, you could control the loop-points with a drumpad such as Push or Launchpad.
In future, this feature could be polished by adding more sophisticated events.

Third event: **move playing cursor to _x_ position**

This one is for randomized compositions (the "probability" in event-roll), and needs to be polished to avoid clicks or un-wanted played notes, so it's just a little harder.

Fourth kind: **link _x_ midi-in to _y_ instrument**

For MIDI drumpad + keyboard live usage.
(of course it would also un-link every other instrument linked to that _x_ midi-in)

Do you agree this would be a worthy addition to LMMS which is not particularly hard to code?
(Unfortunalety I myself lack of indispesable GUI programming knowledge, and I'm yet to understand how note handlers work, but that away, calling existing funcions is the main concept)

**Comment by tresf** [comment]
Closing as this is really a "feeler" post, which is not the intent of a bug tracker.  I agree, that if the enhancement being requested is only simulating mouse clicks, those particular features should be relatively simple to code however this is a relative statement.  Coding, formatting, testing and compiling takes a considerable amount of time for even the smallest feature.

Second, for these user experience type bug reports, we should consider selling them as best we can with mockups.

Here are some posts which illustrate their points very well with pictures:

https://github.com/LMMS/lmms/issues/2214, https://github.com/LMMS/lmms/issues/1950, https://github.com/LMMS/lmms/issues/847, https://github.com/LMMS/lmms/issues/2563

Please use separate bug reports for features which you decide are worthy of writing up.   We don't lock conversations around here, so feel free to use this for Q&A until your questions have been answered.

**Comment by DeRobyJ** [comment]
I'm sorry if it sounded like a "feeler" post, even if I'm new to that term, but I'm really just suggesting a feature. And I was also excusing myself if my previous feature requests were too dreamy. Maybe I should have jumped that part.

I didn't do mockups because I'm not really good at doing them, but I'll try when I'll have more spare time to devote to it.

I know that even a simple feature requires time.

> Please use separate bug reports for features which you decide are worthy of writing up. We don't lock conversations around here, so feel free to use this for Q&A until your questions have been answered.

Err, my bad, but I didn't understand this part. I don't have questions, and this is a feature I think is worthy to have...
Thanks for your concern

**Comment by tresf** [comment]
> Err, my bad, but I didn't understand this part. I don't have questions, and this is a feature I think is worthy to have...

You're right.  I missed the all-encompassing-track portion and dismissed it when I read the opening paragraph.  A mockup will help sell this idea and the whole "live mode" has been beat up for a while... here's a breadcrumb https://github.com/LMMS/lmms/issues/2179.

**Comment by DeRobyJ** [comment]
Ok, next time I'mm go right into the feature xD

I'll do mockups sooner or later, and sorry for deleted post
